gluten-core/src/main/scala/io/glutenproject/extension/RewriteCount.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.extension
+
+import io.glutenproject.backendsapi.BackendsApiManager
+import io.glutenproject.execution.HashAggregateExecTransformerUtil._
+import io.glutenproject.extension.columnar.TransformHints
+
+import org.apache.spark.sql.catalyst.expressions.{If, IsNull, Literal, Or}
+import org.apache.spark.sql.catalyst.expressions.aggregate.{AggregateExpression, Count, Partial}
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.execution.aggregate.{BaseAggregateExec, HashAggregateExec, ObjectHashAggregateExec, SortAggregateExec}
+import org.apache.spark.sql.types.IntegerType
+
+/**
+ * This Rule is used to rewrite the count for aggregates if:
+ *   - the count is partial
+ *   - the count has more than one child
+ *   - the aggregate expressions do not need row construct
+ *
+ * TODO: Remove this rule when Velox support multi-children Count
+ */
+object RewriteCount extends Rule[SparkPlan] {
```

Comment:
Thanks! Generally LGTM. Can we add some notes here about how we rewrite the plan? From my reading, we will use If and isNull to replace the multiple children.

Reply:
sure, added the notes